client/extensions/woocommerce/state/sites/shipping-zones/actions.js

```diff
@@ -34,19 +35,22 @@ export const fetchShippingZones = ( siteId ) => ( dispatch, getState ) => {
 
 	dispatch( getAction );
 
-	return request( siteId ).get( 'shipping/zones' )
-		.catch( err => {
-			dispatch( setError( siteId, getAction, err ) );
-		} )
-		.then( ( data ) => {
-			if ( ! data ) {
-				return;
-			}
-			dispatch( fetchShippingZonesSuccess( siteId, data ) );
-			return Promise.all( data.map( zone => {
-				return fetchShippingZoneMethods( siteId, zone.id )( dispatch, getState );
-			} ).concat( data.map( zone => {
-				return fetchShippingZoneLocations( siteId, zone.id )( dispatch, getState );
-			} ) ) );
-		} );
+	return Promise.all( [
+		request( siteId ).get( 'shipping/zones' ),
+		fetchShippingMethods( siteId )( dispatch, getState ),
```

Comment:
`fetchShippingMethods` already has error handling, so you shouldn't call it at this level.

My gut tells me that fetching the shipping methods should be handled outside of this file. Shipping methods and shipping zones are different parts from the API, which just happen to be used by the same UI. Would it be too messy to call `fetchShippingMethods` where you're already calling `fetchShippingZones`? (And probably change the Spinner condition to `areShippingZonesLoaded() && areShippingMethodsLoaded()`)